js_modules/dagit/packages/core/src/app/__tests__/CustomConfirmationProvider.test.tsx

```diff
@@ -55,22 +57,22 @@ describe('CustomConfirmationProvider', () => {
     );
 
     const button = screen.queryByRole('button', {name: /terminate/i}) as HTMLButtonElement;
-    await userEvent.click(button);
+    userEvent.click(button);
 
-    const confirmationButton = screen.queryByRole('button', {
-      name: /confirm/i,
-    }) as HTMLButtonElement;
-
-    // Resolves the promise.
-    await userEvent.click(confirmationButton);
+    await waitFor(async () => {
+      const confirmationButton = await screen.findByRole('button', {name: /confirm/i});
+      userEvent.click(confirmationButton);
```

Comment:
does it matter that the click is inside the `waitFor` ?

Reply:
No it doesn't -- we could also write it this way, it just repeats the find clause twice. I think I put it inside the block so that I didn't have to copy the regexp:

```

    await waitFor(async () => {
      await screen.findByRole('button', {name: /confirm/i});
    });
    userEvent.click(screen.getByRole('button', {name: /confirm/i}));

 
```

Reply:
you can also write 
```
    const button = await waitFor(async () => screen.findByRole('button', {name: /confirm/i});
    userEvent.click(button);
```

Reply:
Nice! Will switch it over to this format